src/bundle/Command/UpdateUserCommand.php

```diff
@@ -0,0 +1,115 @@
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Core\Repository\Values\User\User;
+use Ibexa\Contracts\Core\Repository\Values\User\UserUpdateStruct;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+#[AsCommand(name: 'ibexa:user:update-user', description: 'Updates basic user data.')]
+final class UpdateUserCommand extends Command
+{
+    public function __construct(
+        private readonly UserService $userService,
+        private readonly Repository $repository,
+        ?string $name = null
+    ) {
+        parent::__construct($name);
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument(
+            'user',
+            InputArgument::REQUIRED,
+            'User login',
+        );
+        $this->addOption(
+            'password',
+            null,
+            InputOption::VALUE_NONE,
+            'New plaintext password (input will be in a "hidden" mode)',
+        );
+        $this->addOption(
+            'email',
+            null,
+            InputOption::VALUE_REQUIRED,
+            'New e-mail address',
+        );
+        $this->addOption(
+            'enable',
+            null,
+            InputOption::VALUE_NONE,
+            'Flag enabling the user being updated',
+        );
+        $this->addOption(
+            'disable',
+            null,
+            InputOption::VALUE_NONE,
+            'Flag disabling the user being updated',
+        );
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $userReference = $input->getArgument('user');
+        $password = $input->getOption('password');
+        $enable = $input->getOption('enable');
+        $disable = $input->getOption('disable');
+        $email = $input->getOption('email');
+
+        if (!$password && !$enable && !$disable && $email === null) {
+            $io->success('No new user data specified, exiting.');
+
+            return Command::SUCCESS;
+        }
+
+        $user = $this->userService->loadUserByLogin($userReference);
+
+        if ($enable && $disable) {
+            $io->error('--enable and --disable options cannot be used simultaneously.');
+
+            return Command::FAILURE;
+        }
+
+        if ($password) {
+            $password = $io->askHidden('Password (your input will be hidden)');
+            $input->setOption('password', $password);
```

Comment:
This does not take into account that Command might be in non-interactive mode.

Default behavior for non-interactive mode is to return `null` (or default). So this will basically not work, crash and burn when trying to execute user update struct.

Additionally, you **may** want to move this to `Command::interact()` method depending on your preferences. The primary difference would be that `interact` is executed before `Command::execute()` (so it might be too early for you) and it's skipped in non-interactive mode. It's primary purpose is to fill any information that might be missing (so it happens before the command validates it's inputs).

Reply:
The only way to handle this I was able to find is https://symfony.com/doc/current/console/input.html#options-with-optional-arguments. Please comment if this works for you:
`bin/console ibexa:user:update-user admin --password=SuperSecurePassword --no-interaction`
vs
`bin/console ibexa:user:update-user admin --password`
the latter one will work with hidden prompt as intended.

Reply:
No, I'd consider using `--password` option and being in a non-interactive mode to be simply an error.

Unless we want to accept password through the option itself. I don't mind it myself (developer is responsible for their security in my opinion, and you could use this for a dev environment within reason), but since we expect it to not be possible, then I'd expect us to error out.

Reply:
The latter is basically what I am proposing right now. I will keep it as-is (usage mentioned above) since you mentioned that's ok from your POV.

Reply:
To put it plainly: if `$input->getOption('password')` gives you something else than `string|null` (because it can give you `true` if `--password` is used with value) then the command should report an error.

Otherwise you pass that boolean value into the struct and rely on later checks for failure - which in turn prevents you from giving the developer/user info that `--password` without value and `--no-interaction` (which can also be enabled via env on CI, btw) are mutually exclusive.